src/mysql_router_helpers.py

```diff
@@ -157,35 +116,34 @@ def bootstrap_and_start_mysql_router(
             bootstrap_mysqlrouter_command.append("--force")
 
         try:
-            subprocess.run(bootstrap_mysqlrouter_command)
+            subprocess.run(bootstrap_mysqlrouter_command, check=True)
 
-            subprocess.run(f"sudo chmod 755 {MYSQL_HOME_DIRECTORY}/{name}".split())
+            replace_socket_location_command = [
+                "sudo",
+                "sed",
+                "-Ei",
+                f"s:/tmp/(.+).sock:{CHARMED_MYSQL_COMMON_DIRECTORY}/var/run/mysqlrouter/\\1.sock:g",
+                f"{CHARMED_MYSQL_DATA_DIRECTORY}/etc/mysqlrouter/mysqlrouter.conf",
+            ]
+            subprocess.run(replace_socket_location_command, check=True)
```

Comment:
@delgod mysqlrouter --bootstrap doesn't allow to specify socket location.
Should we consider to send improvement PR to the vendor or stay with sed forever?